Lib/http/server.py

```diff
@@ -890,17 +891,32 @@ def guess_type(self, path):
         as a default; however it would be permissible (if
         slow) to look inside the data to make a better guess.
 
+        >>> testserver = type('', (), {
+        ...  'makefile': lambda mode, bufsize: open(os.devnull, mode)
+        ... })
+        >>> testhandler = SimpleHTTPRequestHandler(testserver, None, None)
+        >>> testhandler.guess_type('/foo.bar.GZ')  # tests ext.lower()
+        'application/gzip'
+        >>> testhandler.default_content_type = 'nonesuch/nonesuch'
+        >>> testhandler.guess_type('/this/should/give/default')
+        'nonesuch/nonesuch'
+        >>> # check short-circuiting works using mock mimetypes.guess_type
+        >>> sys.modules[__name__].mimetypes = type('', (), {
+        ...  'guess_type': lambda x: (print('foo'), print('bar'))
+        ... })
+        >>> testhandler.guess_type('/should/show/foo/bar/then/default')
+        foo
+        bar
+        'nonesuch/nonesuch'
+        >>> testhandler.guess_type('/should/not/print/foo.Z')
+        'application/octet-stream'
```

Comment:
The docstring of `guess_type` should document that function; it's not a place to add tests for `default_content_type`.
Ideally, we'd have tests for the `python -m http.server` CLI. Unfortunately, we don't have those. Do you want to start them?

API tests for `SimpleHTTPRequestHandler` are in `test_httpservers.py`. You could add this testcase there. (But, please, use a `class` statement to make classes.)

Reply:
OK, I'll fix this as time permits. Thank you.